### DISCUSSION TITLE
wagtail-impersonate for easy impersonation from Users list

### DISCUSSION THREAD

**Discussion by ToJIka4** [wagtail/wagtail#12286] [Show and tell]
Hello everyone!

I've recently started using Wagtail not just as a CMS, but as a framework for building applications. During development and administration, I often need to switch between users for testing purposes. After trying several makeshift solutions, I decided to create a more robust approach.

Check out my new package: [wagtail-impersonate](https://github.com/ToJIka4/wagtail-impersonate). [PyPI](https://pypi.org/project/wagtail-impersonate/).

### How to Use

1. Go to `Settings` -> `Users`.
2. Find the user you want to impersonate.
3. Click on the dots (More options for the user).
4. Click the `Impersonate` button.

To end impersonation, simply click `Stop impersonation` on the admin sidebar.

I hope you find this module useful! Feedback and contributions are welcome.

**Comment by thibaudcolas** [comment]
That’s really cool! I think a screenshot or two in the README would help a lot, in particular I’d love to see how you adapted the UI when impersonation is active.

**Reply by robmoorman** [reply]
Hi @ToJIka4 that's nice! We use https://django-hijack.readthedocs.io/en/latest/ a lot, what is the advantage of using this package?

**Reply by ToJIka4** [reply]
Hi @thibaudcolas! Thanks for the feedback! I’ve updated the README with screenshots showing the UI, including what it looks like when impersonation is active. Let me know if you have any other suggestions!

**Reply by ToJIka4** [reply]
Hi @robmoorman! Thanks for your interest! The main advantage of `wagtail-impersonate` is that it’s specifically tailored for Wagtail, providing a seamless integration with Wagtail’s admin interface. While `django-hijack` is great for general Django projects, `wagtail-impersonate` is built to work directly with Wagtail’s user management and admin workflows, making it easier to use in Wagtail-based applications.
